sdk/core/azure-core/src/datetime.cpp

```diff
@@ -449,33 +456,33 @@ DateTime DateTime::FromString(std::string const& dateString, DateFormat format)
       ++month;
       if (month == 12)
       {
-        return result;
+        throw DateTimeException("Error parsing DateTime: Month number.");
       }
     }
 
     if (str[3] != ' ')
     {
-      return result;
+      throw DateTimeException("Error parsing DateTime.");
     }
 
     str += 4; // parsed month
 
     if (!IsDigit(str[0]) || !IsDigit(str[1]) || !IsDigit(str[2]) || !IsDigit(str[3])
         || str[4] != ' ')
     {
-      return result;
+      throw DateTimeException("Error parsing DateTime: Year.");
     }
 
     int year = (str[0] - '0') * 1000 + (str[1] - '0') * 100 + (str[2] - '0') * 10 + (str[3] - '0');
-    if (year < 1601)
+    if (year < MinYear)
     {
-      return result;
+      throw DateTimeException("Error parsing DateTime: Year is less than 1601.");
```

Comment:
Use constants like `MinYear` within the exception message rather than hard cording it. Same for 9999.

Reply:
to do that, we ill need to bring ostringstream to format messages, I don't think it is worth the cost. It is ok to use constants so that you can distinguish between to different identical values, but it does not mean that you can change 1601 to 0001, or switch the resolution, and expect it to work. You'll need to review all the occurrences, just as we did when we rebesed from year 1900 to 1601. I think this implementation should get rewritten at some point, and there we should think of the most effective ways to make it extensible.